docs/Lua API.rst

```diff
@@ -2984,6 +2984,63 @@ Predefined instance methods:
 
 To avoid confusion, these methods cannot be redefined.
 
+.. _custom-raw-tags:
+
+custom-raw-tags
+===============
+
+A module for reading custom tags added to the raws by mods.
+
+* ``customRawTags.getTag(typeDefinition, tag)``
+
+  Where ``typeDefinition`` is a type definition struct as seen in ``df.global.world.raws``
+  (e.g.: ``dfhack.gui.getSelectedItem().subtype``) and ``tag`` is the name of the custom tag you
+  want to read. The arguments from the tag will then be returned as strings using single or
+  multiple return values. If the tag is not present, the result is false, if it is present
+  but has no arguments, the result is true.
+
+* ``customRawTags.getRaceCasteTag(raceDefinition, casteNumber)``
+
+  Gets tag from a race or caste instead if appplicable, where raceDefinition is of type ``creature_raw`` and casteNumber is an ``int`` (can be ``-1`` for no caste).
+
+* ``customRawTags.getUnitTag(unit, tag)``
+
+  Gets tag from a unit's race or caste instead if applicable. This is a wrapper over ``getRaceCasteTag``.
+
+Examples:
+
+* Using an eventful onReactionComplete hook, something for disturbing dwarven science::
+
+    if customRawTags.getTag(reaction, "CAUSES_INSANITY") then
+        -- make unit who performed reaction go insane
+
+* Using an eventful onProjItemCheckMovement hook, a fast or slow-firing crossbow::
+
+    -- check projectile distance flown is zero, get firer, etc...
+    local multiplier = tonumber(customRawTags.getTag(bow.subtype, "FIRE_RATE_MULTIPLIER")) or 1
+    firer.counters.think_counter = firer.counters.think_counter * multiplier
+
+* Something for a script that prints help text about different types of units::
+
+    local unit = dfhack.gui.getSelectedUnit()
+    if not unit then return end
+    local raceRaw = df.global.world.raws.creatures.all[unit.race]
+    local helpText = customRawTags.getTag(raceRaw, "HELP_TEXT")
+    if helpText then print(helpText) end
+
+* Healing armour::
+
+    -- (per unit every tick)
+    local healAmount = 0
+    for _, entry in ipairs(unit.inventory) do
+        if entry.mode == 2 then -- Worn
+            healAmount = healAmount + tonumber((customRawTags.getTag(entry.item.subtype, "HEAL_AMOUNT")) or 0)
+        end
+    end
+    unit.body.blood_count = math.min(unit.body.blood_max, unit.body.blood_count + healAmount)
+
+The cache is cleared when a map is unloaded.
```

Comment:
since we removed references to the cache above, and since the cache no longer has potentially unexpected side effects, maybe remove this note too.